aws/resource_aws_quicksight_group_test.go

```diff
@@ -0,0 +1,241 @@
+					testAccCheckQuickSightGroupExists(resourceName, &group),
+					resource.TestCheckResourceAttr(resourceName, "description", "Description 1"),
+				),
+			},
+			{
+				Config: testAccAWSQuickSightGroupConfigWithDescription(rName, "Description 2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckQuickSightGroupExists(resourceName, &group),
+					resource.TestCheckResourceAttr(resourceName, "description", "Description 2"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSQuickSightGroup_disappears(t *testing.T) {
+	oldvar := os.Getenv("AWS_DEFAULT_REGION")
+	os.Setenv("AWS_DEFAULT_REGION", "us-east-1")
+	defer os.Setenv("AWS_DEFAULT_REGION", oldvar)
+
+	var group quicksight.Group
+	resourceName := "aws_quicksight_group.default"
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckQuickSightGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSQuickSightGroupConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckQuickSightGroupExists(resourceName, &group),
+					testAccCheckQuickSightGroupDisappears(&group),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccCheckQuickSightGroupExists(resourceName string, group *quicksight.Group) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		awsAccountID, namespace, groupName, err := resourceAwsQuickSightGroupParseID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).quicksightconn
+
+		input := &quicksight.DescribeGroupInput{
+			AwsAccountId: aws.String(awsAccountID),
+			Namespace:    aws.String(namespace),
+			GroupName:    aws.String(groupName),
+		}
+
+		output, err := conn.DescribeGroup(input)
+
+		if err != nil {
+			return err
+		}
+
+		if output.Group == nil {
+			return fmt.Errorf("Quick Sight Group (%s) not found", rs.Primary.ID)
+		}
+
+		*group = *output.Group
+
+		return nil
+	}
+}
+
+func testAccCheckQuickSightGroupDestroy(s *terraform.State) error {
+	meta := testAccProvider.Meta()
+	client := meta.(*AWSClient).quicksightconn
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_quicksight_group" {
+			continue
+		}
+
+		awsAccountID, namespace, groupName, err := resourceAwsQuickSightGroupParseID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		_, err = client.DescribeGroup(&quicksight.DescribeGroupInput{
+			AwsAccountId: aws.String(awsAccountID),
+			Namespace:    aws.String(namespace),
+			GroupName:    aws.String(groupName),
+		})
+		if !isAWSErr(err, quicksight.ErrCodeResourceNotFoundException, "") {
```

Comment:
We generally prefer to split this out so we can display any API errors 👍 

```go
if isAWSErr(err, quicksight.ErrCodeResourceNotFoundException, "") {
	continue
}

if err != nil {
	return err
}

return fmt.Errorf("Quick Sight Group '%s' was not deleted properly", rs.Primary.ID)
```

Reply:
fixed.